src/garage/torch/algos/multi_task_sac.py

```diff
@@ -0,0 +1,246 @@
+        self._epoch_mean_success_rate = []
+        self._epoch_median_success_rate = []
+
+    def _get_log_alpha(self, **kwargs):
+        """Return the value of log_alpha.
+
+        This function exists in case there are versions of MTSAC that need
+        access to a modified log_alpha, such as multi_task MTSAC.
+
+        Args:
+            kwargs(dict): keyword args that can be used in retrieving the
+                log_alpha parameter. Unused here.
+
+        Returns:
+            torch.Tensor: log_alpha
+
+        """
+        obs = kwargs['obs']
+        log_alpha = self.log_alpha
+        one_hots = obs[:, :self._num_tasks]
+        ret = torch.mm(one_hots, log_alpha.unsqueeze(0).t()).squeeze()
+        assert ret.size() == torch.Size([self.buffer_batch_size])
+        return ret
+
+    def optimize_policy(self, itr, samples_data):
+        """Optimize the policy q_functions, and temperature coefficient.
+
+        Args:
+            itr (int): Iterations.
+            samples_data (list): Processed batch data.
+
+        Returns:
+            torch.Tensor: loss from actor/policy network after optimization.
+            torch.Tensor: loss from 1st q-function after optimization.
+            torch.Tensor: loss from 2nd q-function after optimization.
+
+        """
+        obs = samples_data['observation']
+        qf1_loss, qf2_loss = self._critic_objective(
+            samples_data, get_log_alpha_kwargs=dict(obs=obs))
+
+        self.qf1_optimizer.zero_grad()
+        qf1_loss.backward()
+        self.qf1_optimizer.step()
+
+        self.qf2_optimizer.zero_grad()
+        qf2_loss.backward()
+        self.qf2_optimizer.step()
+
+        action_dists = self.policy(obs)
+        new_actions_pre_tanh, new_actions = (
+            action_dists.rsample_with_pre_tanh_value())
+        log_pi_new_actions = action_dists.log_prob(
+            value=new_actions, pre_tanh_value=new_actions_pre_tanh)
+
+        policy_loss = self._actor_objective(obs,
+                                            new_actions,
+                                            log_pi_new_actions,
+                                            get_log_alpha_kwargs=dict(obs=obs))
+        self.policy_optimizer.zero_grad()
+        policy_loss.backward()
+
+        self.policy_optimizer.step()
+
+        if self.use_automatic_entropy_tuning:
+            alpha_loss = self._temperature_objective(
+                log_pi_new_actions, get_log_alpha_kwargs=dict(obs=obs))
+            self.alpha_optimizer.zero_grad()
+            alpha_loss.backward()
+            self.alpha_optimizer.step()
+
+        return policy_loss, qf1_loss, qf2_loss
+
+    def _evaluate_policy(self, epoch):
+        """Evaluate the performance of the policy via deterministic rollouts.
+
+            Statistics such as (average) discounted return and success rate are
+            recorded.
+
+        Args:
+            epoch(int): The current training epoch.
```

Comment:
note that in the format there's a space between then name and type. i'm not sure if the parser correctly handles no-space. you can check.